docs/DeveloperGuide_Winery.md

```diff
@@ -0,0 +1,622 @@
+interface.
+
+##### Project org.eclipse.winery.model.csar.toscametafile
+
+This project contains the model for the file "Tosca.meta".
+
+##### Project org.eclipse.winery.model.selfservice
+
+This project contains the model for the self service portal. It is used by the Vinothek [BBKL14] to display user
+defined data.
+
+##### Project org.eclipse.winery.model.tosca
+
+This project contains a JAX B generated model of the XSD of OASIS TOSCA v1.0. The XSD hat to be modified to
+enable proper referencing and use. An Implementation Artifactmay carry a "name" attribute. The contents of
+properties of Boundary Definitions are processed in "lax" mode.
+
+##### Project org.eclipse.winery.repository.client
+
+Whis project hosts a client using the REST API of the repository and offering a Java object based client to the
+Winery repository.
+
+##### Project org.eclipse.winery.repository
+
+This is the heart of Winery. This project hosts the repository, where all entities of TOSCA are stored and
+managed. It realizes the components "Type, Template, and Artifact Management" and "Repository" (Figure 1).
+
+##### Project org.eclipse.winery.topologymodeler
+
+This part of Winery enables modeling topologies graphically. It realizes the component "Topology Modeler"
+(Figure 1).
+
+---
+
+#### Winery's Id System
+
+The general idea behind the storage of Winery is that each entity comes with an id. The id is either self
+contained or references a parent id, where the element is nested in. All Ids inherit from GenericId.  
+Figure 2 shows the inheritance hierarchy of GenericId. The child "AdminId" is used for all administrative elements
+required for internal management. "DummyParentForGeneratedXSDRef" is required during the export of 
+generated XML Schema Definitions due to the use of Winery's key/value properties. "TOSCAComponentId" is
+the parent element for all TOSCA Elements which may be defined directly as child of a "Definitions" element.
+All other elements have "TOSCAElementId" as parent.
+
+<center>
+
+![alt text](graphics/Figure02-InheritanceHierarchyOfGenericId.png "Figure 2: Inheritance hierarchy of GenericId")  
+Figure 2: Inheritance hierarchy of GenericId
+
+</center> 
+
+##### AdminId
+
+Figure 3 shows the inheritance hierarchy of AdminId. "NamespacesId" is used as container for a mapping file
+from namespace prefixes to namespaces. "TypesId" is the parent element of all types user can set. This are 
+not node types etc., but ConstraintTypes (for Constraints), PlanLanguages (for plans), and PlanTypes (for plans).
+The inclusion of "PlanLanguages" is due to reuse of the class AbstractTypesManager for plan languages. TOSCA
+does not restrict these enumerations. Therefore, Winery manages all known types for itself.
+
+<center>
+
+![alt text](graphics/Figure03-IneritanceHierarchyofAdminId.png "Figure 3: Inheritance hierarchy of AdminId")  
+Figure 3: Inheritance hierarchy of AdminId
```

Comment:
Please use the same format for captions throughout the document. Above, it was bold, how it is plain text.